### PR TITLE
Combine validate and build in to a single operation

### PR DESCRIPTION
**Pull request by mpilquist**
Incorporating some ideas from @martijnhoekstra

### CODE REVIEW

File: tests/shared/src/main/scala-3/org/typelevel/literally/examples/literals.scala [hunk 1]
```diff
@@ -28,18 +28,12 @@ object literals:
...
-    def build(s: String)(using Quotes) =
-      '{ShortString.unsafeFromString(${Expr(s)})}
+    def validate(s: String)(using Quotes) =
+      if s.length <= ShortString.MaxLength then Right('{ShortString.unsafeFromString(${Expr(s)})}) 
+      else Left(s"ShortString must be <= ${ShortString.MaxLength} characters")
 
   object PortLiteral extends Literally[Port]:
-    def validate(s: String): Option[String] =
+    def validate(s: String)(using Quotes) =
       Try(s.toInt).toOption.flatMap(Port.fromInt) match
```
Comment (martijnhoekstra) [inline tests/shared/src/main/scala-3/org/typelevel/literally/examples/literals.scala:37]: `toIntOption`?

Comment (mpilquist) [reply tests/shared/src/main/scala-3/org/typelevel/literally/examples/literals.scala:37]: Ha, I didn't use that b/c this project cross compiles for 2.12 but this file is Scala 3 only so I'll update. :)

Comment (smarter) [inline tests/shared/src/main/scala-3/org/typelevel/literally/examples/literals.scala]: The 2.13 standard library has toIntOption

Comment (mpilquist) [reply tests/shared/src/main/scala-3/org/typelevel/literally/examples/literals.scala]: Yep - the scala-2 source file builds for 2.12 still though

**Pull request merged**